### PR TITLE
Fix review conversation reply

### LINKED ISSUE #23843: Reply in the conversation of pull request UI generated a review message.
I think the behavior is changed from some PR. It's not the same as Github and make me confusing.

### PR DESCRIPTION
**Pull request by wxiaoguang**
Regression of #23245

Close #23843

### CODE REVIEW

**Comment by silverwind** [comment]
1.19 unaffected, right?

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> 1.19 unaffected, right?

Nope, no backport

File: templates/repo/diff/comment_form.tmpl [hunk 1]
```diff
@@ -27,6 +27,7 @@
 				{{if $.reply}}
 					<button class="ui submit green tiny button btn-reply" type="submit">{{$.root.locale.Tr "repo.diff.comment.reply"}}</button>
 					<input type="hidden" name="reply" value="{{$.reply}}">
+					<input type="hidden" name="single_review" value="true">
```
Comment (delvh) [inline templates/repo/diff/comment_form.tmpl:30]: I have a slight feeling that the current mechanism will always be somewhat erroneous:

- With your proposed approach, it won't be possible to use replies as part of a pending review.
The question is whether we want to allow that or not…
- With the current approach, replies are listed as a separate review.

Comment (wxiaoguang) [reply templates/repo/diff/comment_form.tmpl:30]: That's the old behavior before the regression. At that time, a default form submission means comment but not pending.

----

~~Or maybe I made something wrong, but I haven't figured out. I felt that this fix works like before~~

Update: I think this fix is not wrong, see comment below.

Comment (wxiaoguang) [reply templates/repo/diff/comment_form.tmpl:30]: Or maybe I made something wrong, but I haven't figured out. I felt that this fix works like before

Comment (wxiaoguang) [reply templates/repo/diff/comment_form.tmpl:30]: * "With your proposed approach"
    That's not my proposal, that's how it worked before. Actually I think after my refactoring, the logic is slightly clearer than before, it's easier to add a correct pending behavior in the future.


* "With the current approach, replies are listed as a separate review."
     And confirmed with Gitea 1.16, it was already like that before

So this PR only fixes the "pending" problem, I think we should leave other improvements to the future.

Screenshot of 1.16

<details>

image

</details>

Screenshot of 1.20

<details>

image

</details>

**Review by lunny — APPROVED** [review]

**Pull request merged**